components/Login/LoginModal.js

```diff
@@ -65,18 +64,30 @@ class LoginModal extends React.Component {
         return false;
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.UserProfileStore.errorMessage !== '') {
-            $('.ui.form.signin').form('add errors', [nextProps.UserProfileStore.errorMessage]);
+    handleRegisterFirst(dismiss) {
+        localStorage.setItem(MODI, 'login_failed_register_now');
+
+        let thatContext = this.context;
+        async.series([
+            function(callback) {
+                thatContext.executeAction(navigateAction, {
+                    url: '/signup'
+                });
+                callback(null, 'two');
+            }
+        ]);
+
+        return true;
+    }
+
+    componentDidUpdate() {
+        // console.log('componentDidUpdate:', this.props.errorMessage, this.props.socialLoginError, this.props.userid, this.props.username);
+        if (this.props.errorMessage !== '' && this.props.errorMessage !== undefined && this.isLoading) {
```

Comment:
You could use `!common.isEmpty()` instead of these two statements `this.props.errorMessage !== '' && this.props.errorMessage !== undefined`